sdk/spring/azure-spring-integration-servicebus/src/main/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverter.java

```diff
@@ -115,7 +115,6 @@ protected void setCustomHeaders(MessageHeaders headers, ServiceBusMessage messag
         getAndRemove(copySpringMessageHeaders, CORRELATION_ID).ifPresent(message::setCorrelationId);
         getAndRemove(copySpringMessageHeaders, TO).ifPresent(message::setTo);
         getAndRemove(copySpringMessageHeaders, REPLY_TO_SESSION_ID).ifPresent(message::setReplyToSessionId);
-        getAndRemove(copySpringMessageHeaders, PARTITION_KEY).ifPresent(message::setPartitionKey);
```

Comment:
How about handling all possible values of partitionkey and put its final value in spring message headers in function com.azure.spring.integration.servicebus.ServiceBusTemplate#getPartitionKey, and then in this function of converting message, we could read the associated header directly?